rpc/wrtc_signaling_answerer.go

```diff
@@ -194,19 +194,39 @@ func (ans *webrtcSignalingAnswerer) startAnswerer() {
 			default:
 			}
 			var err error
+			// `newAnswer` opens a bidi grpc stream to the signaling server. But otherwise sends no requests.
 			client, err = newAnswer()
+			receivedInitRequest := false
 			if err == nil {
-				err = ans.answer(client)
+				// `ans.answer` will send the initial message to the signaling server that says it
+				// is ready to accept connections. Then it waits, typically for a long time, for a
+				// caller to show up. Which is when the signaling server will send a
+				// response. `ans.answer` then follows with gathering ICE candidates and learning of
+				// the caller's ICE candidates to create a working WebRTC PeerConnection. If
+				// successful, the `PeerConnection` + `webrtcServerChannel` will be registered and
+				// available for the `webrtcServer`.
+				receivedInitRequest, err = ans.answer(client)
 			}
-			// Exceptional errors represent a broken connection. While direct gRPC connections will reconnect
-			// on their own, we should wait a little before trying to call again. Common
-			// errors represent that an operation has failed, but can be safely retried over the
-			// existing connection.
-			if checkExceptionalError(err) != nil {
-				ans.logger.Errorw("error answering", "error", err)
+
+			switch {
+			case err == nil:
+			case receivedInitRequest && err != nil:
+				// We received an error while trying to connect to a caller/peer.
+				ans.logger.Errorw("error connecting", "error", err)
```

Comment:
A little confused about these cases. If we've received the init request, doesn't that mean we had an `error answering` and _not_ an `error connecting`?

Reply:
I think you and I have different ideas what connecting means. Will try different words